### PR TITLE
Update common files

### PR DESCRIPTION
**Pull request by carolynhu**
https://github.com/istio/istio/issues/18173

In istio/operator, we introduced a change in this PR: Add vars for hub and tag that can be populated during build time: https://github.com/istio/operator/pull/416
We set build variables in `istio/common/scripts/report_build_info.sh`. 
This PR is to add the HUB and TAG variables when we set build variables.

### CODE REVIEW

File: common/scripts/report_build_info.sh [hunk 1]
```diff
@@ -38,14 +38,17 @@ fi
 
 # security wanted VERSION='unknown'
 VERSION="${BUILD_GIT_REVISION}"
-if [[ -n ${ISTIO_VERSION} ]]; then
-  VERSION="${ISTIO_VERSION}"
+if [[ -n ${TAG_VARIANT} ]]; then
+  VERSION="${TAG_VARIANT}"
```
Comment (howardjohn) [inline common/scripts/report_build_info.sh]: this should stay as ISTIO_VERSION I think. What was the reason for changing it?

File: common/scripts/report_build_info.sh [hunk 2]
```diff
@@ -38,14 +38,17 @@ fi
...
 fi
 
 GIT_DESCRIBE_TAG=$(git describe --tags)
+HUB=""
```
Comment (howardjohn) [inline common/scripts/report_build_info.sh]: This means HUB is always `""`. We should instead set `HUB=${HUB:-""}`, so it can be overriden

**Comment by carolynhu** [comment]
@howardjohn ok, will redirect to common-files

**Comment by carolynhu** [comment]
cc @howardjohn

**Comment by howardjohn** [comment]
Needs some other changes, will be done in https://github.com/istio/istio/pull/18279. Thanks!

**Pull request closed without merging**